tests/services/test_data_svc.py

```diff
@@ -92,3 +167,72 @@ def test_no_autogen_cleanup_cmds(self, event_loop, data_svc):
+        mock_ext_hooks.assert_has_calls([
+            call(dict()),
+            call(PAYLOAD_CONFIG_YAMLS['path1'][0]['extensions']),
+        ], any_order=True)
+
+        expected_config_part1 = {
+            'standard_payloads': {
+                'file.exe': PAYLOAD_CONFIG_YAMLS['path1'][0]['standard_payloads']['file.exe'],
+                'file.ps1': PAYLOAD_CONFIG_YAMLS['path1'][0]['standard_payloads']['file.ps1'],
+                'file.py': PAYLOAD_CONFIG_YAMLS['path2'][0]['standard_payloads']['file.py'],
+                'file.txt': PAYLOAD_CONFIG_YAMLS['path2'][0]['standard_payloads']['file.txt'],
+            },
+            'special_payloads': {
+                'file.go': PAYLOAD_CONFIG_YAMLS['path1'][0]['special_payloads']['file.go'],
+                'file.cpp': PAYLOAD_CONFIG_YAMLS['path2'][0]['special_payloads']['file.cpp'],
+            },
+            'extensions': {
+                '.donut': PAYLOAD_CONFIG_YAMLS['path1'][0]['extensions']['.donut'],
+            }
+        }
+        mock_apply_config.assert_called_once_with(name='payloads', config=expected_config_part1)
+
+        with patch.object(glob, 'iglob', return_value=['path3']) as mock_iglob2:
+            with patch.object(BaseWorld, 'apply_config', wraps=_mock_apply_payload_config) as mock_apply_config2:
+                with patch.object(DataService, 'get_config', return_value=self.mock_payload_config):
+                    event_loop.run_until_complete(data_svc._load_payloads(test_plugin2))
+        mock_iglob2.assert_called_once_with('test_plugin2/data/payloads/*.yml', recursive=False)
+        mock_strip_yml.assert_called_with('path3')
+        mock_payload_hooks.assert_called_with(PAYLOAD_CONFIG_YAMLS['path3'][0]['special_payloads'])
+        mock_ext_hooks.assert_called_with(PAYLOAD_CONFIG_YAMLS['path3'][0]['extensions'])
+
+        expected_config_part2 = {
+            'standard_payloads': {
+                'file.exe': PAYLOAD_CONFIG_YAMLS['path1'][0]['standard_payloads']['file.exe'],
+                'file.ps1': PAYLOAD_CONFIG_YAMLS['path1'][0]['standard_payloads']['file.ps1'],
+                'file.py': PAYLOAD_CONFIG_YAMLS['path2'][0]['standard_payloads']['file.py'],
+                'file.txt': PAYLOAD_CONFIG_YAMLS['path2'][0]['standard_payloads']['file.txt'],
+            },
+            'special_payloads': {
+                'file.go': PAYLOAD_CONFIG_YAMLS['path1'][0]['special_payloads']['file.go'],
+                'file.cpp': PAYLOAD_CONFIG_YAMLS['path2'][0]['special_payloads']['file.cpp'],
+                'special.py': PAYLOAD_CONFIG_YAMLS['path3'][0]['special_payloads']['special.py'],
+            },
+            'extensions': {
+                '.donut': PAYLOAD_CONFIG_YAMLS['path1'][0]['extensions']['.donut'],
+                '.testext': PAYLOAD_CONFIG_YAMLS['path3'][0]['extensions']['.testext'],
+            }
+        }
+        mock_apply_config2.assert_called_once_with(name='payloads', config=expected_config_part2)
```

Comment:
Do we need to test what happens if two plugins have the same payload but maybe a different id?

Reply:
I haven't considered that edge case - what do we want as the outcome? The most recently processed plugin overrides previous entries?

Reply:
Will "most recently processed" be consistent? Are they in alphabetical order or random?

Probably also want to issue an INFO or DEBUG message about it, to help point users to what happened if they expected one thing but got another.

Reply:
Added some checks and warns before override. Adjusted test to include an override example